src/main/core/worker.c

```diff
@@ -510,14 +510,14 @@ gboolean worker_scheduleTask(Task* task, Host* host, SimulationTime nanoDelay) {
 
 EmulatedTime worker_maxEventRunaheadTime(Host* host) {
     utility_assert(host);
-    SimulationTime max = _worker_getRoundEndTime();
+    EmulatedTime max = _worker_getRoundEndTime() + EMULATED_TIME_OFFSET;
 
-    Event* event = scheduler_peekHost(_worker_pool()->scheduler, host);
-    if (event) {
-        max = MIN(max, event_getTime(event));
+    EmulatedTime nextEventTime = scheduler_nextHostEventTime(_worker_pool()->scheduler, host);
+    if (nextEventTime) {
+        max = MIN(max, nextEventTime);
     }
```

Comment:
Maybe add a comment here that 0 is not expected to be a valid `nextEventTime` emulated time since the offset `EMULATED_TIME_OFFSET` is greater than 0 (there will never be an event at an emulated time of 0)?

Reply:
Added a comment to `scheduler_nextHostEventTime` and made the comparison to 0 here explicit